iChor/iChorCNA_hg19/iChorCNA_hg19.wdl

```diff
@@ -1,327 +1,349 @@
-		--txnE ${txnE} \
-		--txnStrength ${txnStrength} \
-        --minSegmentBins ${minSegmentBins} \
-		--minMapScore ${minMapScore} \
-        --lambdaScaleHyperParam ${lambdaScaleHyperParam} \
-		--fracReadsInChrYForMale ${fracReadsChrYMale} \
-		--maxFracGenomeSubclone ${maxFracGenomeSubclone} \
-        --altFracThreshold ${altFracThreshold} \
-		--maxFracCNASubclone ${maxFracCNASubclone} \
-        --rmCentromereFlankLength ${rmCentromereFlankLength} \
-		--plotFileType ${plotFileType} \
-		--plotYLim "${plotYlim}"
-
-		# Zip optimal solutions
-		mkdir ${sample_id}.optimalSolution
-		cp ${sample_id}/${sample_id}_genomeWide.pdf ${sample_id}.cna.seg ${sample_id}.seg.txt ${sample_id}.seg ${sample_id}.optimalSolution/
-		zip -r ${sample_id}.optimalSolution.zip ${sample_id}.optimalSolution
-        
+    input {
+        File wig_file
+        File? normalPanel
+        String genomeBuild
+        Float? mean_depth
+        String sample_id
+        String ploidy
+        String normal
+        Int maxCN
+        Boolean includeHOMD
+        String chrs
+        String chrTrain
+        String chrNormalize
+        String genomeStyle
+        Boolean estimateNormal
+        Boolean estimatePloidy
+        Boolean estimateClonality
+        String scStates
+        File centromere
+        File? exons
+        Float txnE
+        Int txnStrength
+        Int minSegmentBins
+        Float minMapScore
+        Float fracReadsChrYMale
+        Float maxFracCNASubclone
+        Float maxFracGenomeSubclone
+        Float altFracThreshold
+        Int lambdaScaleHyperParam
+        Int rmCentromereFlankLength
+        String plotFileType = "pdf"
+        String plotYlim
+        File input_gcWig
+        File input_mapWig
+
+        Int memGB = 4
+        Int diskGB = 50
+        Int preemptible = 2
+        Int maxRetries = 1
+
+        String docker_override = "us.gcr.io/tag-team-160914/bloodbiopsy-ichorcna:0.2.1"
+    }
+
+    command <<<
+        set -euxo pipefail
+
+        Rscript /runIchorCNA.R --id "~{sample_id}" \
+            --outDir ./ --libdir /ichorCNA \
+            --WIG "~{wig_file}" \
+            --gcWig  "~{input_gcWig}" \
+            --mapWig "~{input_mapWig}" \
+            --normalPanel "~{default="None" normalPanel}" \
+            --ploidy "~{ploidy}" \
+            --normal "~{normal}" \
+            ~{"--coverage " + mean_depth} \
+            --maxCN ~{maxCN} \
+            --includeHOMD ~{includeHOMD} \
+            --chrs "~{chrs}" \
+            --chrTrain "~{chrTrain}" \
+            --chrNormalize "~{chrNormalize}" \
+            --genomeStyle "~{genomeStyle}" \
+            --genomeBuild "~{genomeBuild}" \
+            --estimateNormal ~{estimateNormal} \
+            --estimatePloidy ~{estimatePloidy}  \
+            --estimateScPrevalence ~{estimateClonality} \
+            --scStates "~{scStates}" \
+            --centromere ~{centromere} \
+            ~{"--exons.bed '" + exons + "'"} \
```

Comment:
@geoffjentry I'm not 100% sure about this expression in WDL 1.0. I want the string enclosing which is why I made the change, but to me it's not obvious if adding a _suffix_ will cause an Optional value to remain None. Later WDL specs are clear on this.

Reply:
If you're worried about this postscript change we could
a) revert: although it's possible that an exons file with spaces will crash in hard-to-debug ways
b) test in a simple workflow (hello-world)
c) wrap in something horrible that will definitely succeed, e.g. `~{if(defined(exons)) then "--exons.bed '"+ exons + "'" else ""}`

Reply:
I coulda sworn I responded to the original comment but apparently not: I'd prefer to stick to things that are 100% sure. That doesn't preclude option `b` if you want to go that route.

Reply:
I made a test WDL with `version 1.0` and ran it locally, and it worked as expected.